### PR TITLE
refactor: added connection constructor instead onInit method

### PR DESCRIPTION
**Pull request by KonstantinDM**

### CODE REVIEW

File: lib/pages/edit_page/bloc/edit_page_state.dart [hunk 1]
```diff
@@ -2,6 +2,13 @@ part of 'edit_page_bloc.dart';
 
 @immutable
 class EditPageState extends Equatable {
+  EditPageState.connection(Connection connection)
+      : name = connection.name,
+        accessKey = connection.accessKey,
+        secretKey = connection.secretKey,
+        endpoint = connection.endpoint,
+        bucket = connection.bucket;
+
```
Comment (AlexEbenrode) [inline lib/pages/edit_page/bloc/edit_page_state.dart:11]: Why another constructor?

File: lib/pages/edit_page/bloc/edit_page_bloc.dart [hunk 2]
```diff
@@ -8,15 +8,15 @@ part 'edit_page_event.dart';
 part 'edit_page_state.dart';
 
 class EditPageBloc extends Bloc<EditPageEvent, EditPageState> {
-  EditPageBloc({required this.storage}) : super(const EditPageState()) {
+  EditPageBloc(Connection connection, {required this.storage})
+      : super(EditPageState.connection(connection)) {
```
Comment (AlexEbenrode) [inline lib/pages/edit_page/bloc/edit_page_bloc.dart:12]: Maybe better remove constructor and use previous constructor?
```suggestion
  EditPageBloc(Connection connection, {required this.storage})
      : super(EditPageState(
            name: connection.name,
            accessKey: connection.accessKey,
            secretKey: connection.secretKey,
            bucket: connection.bucket,
            endpoint: connection.endpoint)) {
```

What do you think?

File: lib/pages/edit_page/bloc/edit_page_state.dart (hunk 1 above)
Comment (KonstantinDM) [reply lib/pages/edit_page/bloc/edit_page_state.dart:11]: In order not to break the old semantics and explicitly determine that the state is created based on an existing connection.

**Review by AlexEbenrode — APPROVED** [review]

**Comment by goshander** [comment]
:tada: this issue has been resolved in version 1.1.0 :tada:

release is available on [GitHub](<github_release_url>)

**Pull request merged**